### ISSUE TITLE
Add `curryN` for more explicit partial application 

### ISSUE THREAD

**Issue by evilsoft** [evilsoft/crocks#10]
While the current recursive `curry` function is great in most cases, it would be nice to have the ability to explicitly specify the expected arity of a function. This is handy for complicated compositions with curried mutli-ary functions.

**Comment by flintinatux** [comment]
Wanted to add some details to this one.  I tried an operation like this on the `ramda` repl, and was disappointed by [the result](https://goo.gl/ObnDGO).  A simplified version of the use case I was aiming for looks like this:
```js
const f = curryN(3, op => (dt, s) => [op, dt, s])
f('del', 250, 'foo')  // => should return ['del', 250, 'foo']
```
The best that `ramda` was able to do was curry the outer function, and once passed three arguments, only return the inner function:
```js
function (dt, s) {
  return [op, dt, s];
}
```

**Comment by evilsoft** [comment]
Submitted a [PR#20](https://github.com/evilsoft/crocks/pull/20) to address this issue.
This function cannot have the magic of `curry` auto currying until a non function value is returned.
Instead it is to be used with the various n-ary functions that exist in the wild.

Please see the PR for some examples of usage.

**Comment by evilsoft** [comment]
Merged in [PR#20](https://github.com/evilsoft/crocks/pull/20).
Closing.
